### PR TITLE
Propize task component

### PR DESCRIPTION
**Pull request by kinow**
This PR adds props to the `Task` component, and also updates the current suite tree view to use this new component :+1:

### CODE REVIEW

File: src/components/cylc/Task.vue [hunk 1]
```diff
@@ -158,10 +158,16 @@
...
         name: 'Task',
-        data: function () {
-            return {
-                'status': 'running',
-                'progress': 25
+        props: {
+            status: {
+                type: String,
+                required: true,
+                default: 'waiting'
```
Comment (oliver-sanders) [inline src/components/cylc/Task.vue]: *Purely out of academic interest* is there a way of using an enum from this, I think we can extract the allowed values using introspection on GraphQL.

Comment (kinow) [reply src/components/cylc/Task.vue]: That's in my browser history from yesterday. Something like 'vue enum props component'. Unfortunately there is no clear solution. Some people recommended `type: Function` or `Object` if I recall correctly, then you would have a bit more of control.

A good improvement. Do you think we should create an issue for this, or add a TODO here?

Comment (oliver-sanders) [reply src/components/cylc/Task.vue]: We can leave this as is, a nice improvement but not an important one.

**Comment by kinow** [comment]
Great comment as always @sadielbartholomew, and we don't have any hurry to merge this one, so any constructive discussion is very welcome IMHO. 

>So could we add a note to emphasise it is the status of the Task component, not the task state? Not a blocker, but I think it is an easy assumption to make given the similarity in status & state as descriptors.

Sure, would it be just a comment? But the comment would just be something like `<-- This is the task status, not the task state -->`?

I haven't thought much about it, to be honest, as @oliver-sanders did all the hard work to create the component. His initial work had state and progress, so this PR was more to move them from data values, to props that can be passed to the component.

>Then I was confused as to why it was safe & wise to (as on new line 165) set a default of 'waiting'

I did think a little about this one... but I couldn't come up with a better alternative. What about an empty string then? Not too sure, but I think it would render an empty element, without any CSS class, as the status name is [used to match the CSS style](https://github.com/cylc/cylc-ui/blob/e68f91b9410b5508c8e0fb6f780a9c765ed4638e/src/components/cylc/Task.vue#L15). WDYT?

**Comment by oliver-sanders** [comment]
> I did think a little about this one... but I couldn't come up with a better alternative.

I think the default state value should be blank (`''`) as `waiting` isn't a safe default value (FYI we actually do this in the current GUIs [the "blank" state shows up in black] though you would only see the effects if something went really wrong).

File: src/views/Suite.vue [hunk 2]
```diff
@@ -31,6 +31,9 @@
                 <i class="mdi mdi-plus" v-else></i>
               </span>
             </template>
+            <template slot="state" slot-scope="props">
+              <task :status="props.row.state" :progress=0 />
```
Comment (oliver-sanders) [inline src/views/Suite.vue]: Cool, I diddn't realise it was this easy to propagate properties downstream.

**Comment by kinow** [comment]
Rebased fixing conflicts, and then removed the default `waiting` after I realized it was marked already as `required`, so no need to have a default value :grimacing:

**Pull request merged**